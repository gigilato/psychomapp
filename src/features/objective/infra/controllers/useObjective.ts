import { useQuery } from '@tanstack/react-query'

import { getObjective } from '../fetchers'

export const getQueryKey = (id: string) => ['objectives', id]
export const useObjective = (id?: string) => {
  const query = useQuery({
    queryKey: getQueryKey(id!),
    queryFn: async () => {
      return getObjective(id!)
    },
    enabled: !!id,
  })
  return query
}
