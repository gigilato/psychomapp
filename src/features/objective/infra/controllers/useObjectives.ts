import { useQuery } from '@tanstack/react-query'

import { getObjectives } from '../fetchers'

export const queryKey = ['objectives']
export const useObjectives = () => {
  const query = useQuery({
    queryKey,
    queryFn: async () => {
      return getObjectives()
    },
  })
  return query
}
