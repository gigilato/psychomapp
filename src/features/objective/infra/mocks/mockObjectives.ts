import { Objective } from '$types/database'

export const mockObjectiveList: Objective[] = [
  {
    id: '1',
    name: 'Obj1',
    slug: 'Obj1',
    color: '#1fff00',
    createdAt: '',
    updatedAt: '',
    practitionnerId: '',
  },
  {
    id: '2',
    name: 'Obj2',
    slug: 'Obj2',
    color: '#ff3f00',
    createdAt: '',
    updatedAt: '',
    practitionnerId: '',
  },
  {
    id: '3',
    name: 'Obj3',
    slug: 'Obj3',
    color: '#ffff00',
    createdAt: '',
    updatedAt: '',
    practitionnerId: '',
  },
  {
    id: '4',
    name: 'Obj4',
    slug: 'Obj4',
    color: '#100f50',
    createdAt: '',
    updatedAt: '',
    practitionnerId: '',
  },
]
