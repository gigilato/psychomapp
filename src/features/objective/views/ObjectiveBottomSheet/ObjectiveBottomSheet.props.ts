import { IBottomSheetProps } from '$organisms'

export interface IObjectiveBottomSheetProps extends Omit<IBottomSheetProps, 'children'> {}
