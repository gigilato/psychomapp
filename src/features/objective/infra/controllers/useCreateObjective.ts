import { useMutation, useQueryClient } from '@tanstack/react-query'

import { i18n } from '$infra/i18n'
import { showToast } from '$infra/toast'

import { queryKey } from './useObjectives'
import { CreateObjectiveVariables, createObjective, getObjective } from '../fetchers'

export const useCreateObjective = () => {
  const queryClient = useQueryClient()

  const mutation = useMutation({
    mutationFn: async (variables: CreateObjectiveVariables) => {
      const { result, data } = await createObjective(variables)
      if (result.status !== 201) throw new Error('CreateObjectiveError')
      const objective = await getObjective(data.id)
      return objective
    },
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey })
    },
    onError: () => {
      showToast(i18n.t('errors.default'))
    },
  })
  return mutation
}
