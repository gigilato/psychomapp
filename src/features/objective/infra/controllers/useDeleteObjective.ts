import { useMutation, useQueryClient } from '@tanstack/react-query'

import { i18n } from '$infra/i18n'
import { showToast } from '$infra/toast'
import { Objective } from '$types/database'

import { queryKey } from './useObjectives'
import { DeleteObjectiveVariables, deleteObjective } from '../fetchers'

export const useDeleteObjective = () => {
  const queryClient = useQueryClient()

  const mutation = useMutation({
    mutationFn: async (variables: DeleteObjectiveVariables) => {
      const result = await deleteObjective(variables.id)
      if (result.status !== 204) throw new Error('DeleteObjectiveError')
      return variables
    },
    onMutate: async (variables) => {
      await queryClient.cancelQueries({ queryKey })
      const previous = queryClient.getQueryData<Objective[]>(queryKey)
      queryClient.setQueryData<Objective[]>(
        queryKey,
        (current) => current?.filter((item) => item.id !== variables.id)
      )
      return { previous }
    },
    onError: (_, variables, context) => {
      if (context?.previous) queryClient.setQueryData<Objective[]>(queryKey, context.previous)
      showToast(i18n.t('errors.default'))
    },
    onSettled: () => {
      queryClient.invalidateQueries({ queryKey })
    },
  })
  return mutation
}
