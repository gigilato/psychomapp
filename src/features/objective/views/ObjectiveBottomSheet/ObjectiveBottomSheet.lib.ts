import { BottomSheetModal } from '@gorhom/bottom-sheet'
import { ForwardedRef, useImperativeHandle, useRef } from 'react'
import { Alert } from 'react-native'

import { useCreateObjective } from '$features/objective/infra/controllers/useCreateObjectives'
import { i18n } from '$infra/i18n'

export const useObjectiveBottomSheet = (ref: ForwardedRef<BottomSheetModal>) => {
  const innerRef = useRef<BottomSheetModal>(null)
  useImperativeHandle(ref, () => (innerRef as any).current)

  const { mutate, isPending } = useCreateObjective()
  const onPress = (name?: string) => {
    if (!name || name === '') return
    mutate({ name })
  }
  const onPressCreate = () => {
    Alert.prompt(i18n.t('objectives.create'), undefined, [
      { style: 'cancel', text: i18n.t('common.cancel') },
      { onPress, text: i18n.t('common.validate') },
    ])
  }
  return { innerRef, onPressCreate, isPending }
}
