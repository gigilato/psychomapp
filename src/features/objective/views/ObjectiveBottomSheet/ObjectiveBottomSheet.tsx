import { BottomSheetModal } from '@gorhom/bottom-sheet'
import { forwardRef } from 'react'
import { ActivityIndicator } from 'react-native'

import { Box, Center, Text } from '$atoms'
import { ObjectiveList } from '$features/objective/views/ObjectiveList'
import { PressableText } from '$molecules'
import { BottomSheet } from '$organisms'
import { i18n } from '$shared/infra/i18n/i18n'

import { useObjectiveBottomSheet } from './ObjectiveBottomSheet.lib'
import { IObjectiveBottomSheetProps } from './ObjectiveBottomSheet.props'

export const ObjectiveBottomSheet = forwardRef<BottomSheetModal, IObjectiveBottomSheetProps>(
  (props, ref) => {
    const { innerRef, onPressCreate, isPending, onPressObjective } = useObjectiveBottomSheet(ref)
    return (
      <BottomSheet ref={innerRef} {...props}>
        <Box className="px-l py-m">
          <Text variant="subtitle" className="text-center">
            {i18n.t('objectives.title')}
          </Text>
          <ObjectiveList className="my-l" onPressObjective={onPressObjective} />
          <Center>
            {!isPending ? (
              <PressableText ID="objectives/create" variant="link" onPress={onPressCreate}>
                {i18n.t('objectives.create')}
              </PressableText>
            ) : (
              <ActivityIndicator />
            )}
          </Center>
        </Box>
      </BottomSheet>
    )
  }
)
