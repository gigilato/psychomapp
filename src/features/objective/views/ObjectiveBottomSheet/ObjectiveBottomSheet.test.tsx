import { render } from '@testing-library/react-native'

import { useObjectives } from '$features/objective/infra/controllers/useObjectives'
import { mockObjectiveList } from '$features/objective/infra/mocks/mockObjectives'
import { QueryWrapper } from '$jest/jest.wrappers'

import { ObjectiveBottomSheet } from './ObjectiveBottomSheet'

jest.mock('$features/objective/infra/controllers/useObjectives', () => ({
  useObjectives: jest.fn(),
}))

describe('ObjectiveBottomSheet', () => {
  it('should render correctly', async () => {
    ;(useObjectives as jest.Mock).mockImplementation(() => ({
      data: mockObjectiveList,
      isLoading: false,
      error: false,
    }))

    const Component = render(<ObjectiveBottomSheet />, { wrapper: QueryWrapper })

    expect(Component).toMatchSnapshot()
  })

  it('should render loading', async () => {
    ;(useObjectives as jest.Mock).mockImplementation(() => ({
      data: undefined,
      isLoading: true,
      error: false,
    }))

    const Component = render(<ObjectiveBottomSheet />, { wrapper: QueryWrapper })

    expect(Component).toMatchSnapshot()
  })

  it('should render error', async () => {
    ;(useObjectives as jest.Mock).mockImplementation(() => ({
      data: undefined,
      isLoading: false,
      error: true,
    }))

    const Component = render(<ObjectiveBottomSheet />, { wrapper: QueryWrapper })

    expect(Component).toMatchSnapshot()
  })
})
