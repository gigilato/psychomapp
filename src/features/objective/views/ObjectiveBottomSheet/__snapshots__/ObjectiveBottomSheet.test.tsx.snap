// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectiveBottomSheet should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "paddingLeft": 24,
          "paddingRight": 24,
        },
        {
          "paddingBottom": 16,
          "paddingTop": 16,
        },
        [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Bold",
          },
          {
            "fontSize": 20,
          },
          {
            "lineHeight": 28,
          },
          {
            "textAlign": "center",
          },
        ]
      }
    >
      Mes objectifs
    </Text>
    <View
      style={
        [
          {
            "marginBottom": 24,
            "marginTop": 24,
          },
          [
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "flexWrap": "wrap",
            },
            [
              undefined,
              [
                {
                  "marginLeft": -8,
                  "marginTop": -8,
                },
                undefined,
              ],
            ],
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "borderBottomLeftRadius": 9999,
                "borderBottomRightRadius": 9999,
                "borderTopLeftRadius": 9999,
                "borderTopRightRadius": 9999,
              },
              {
                "paddingBottom": 4,
                "paddingLeft": 4,
                "paddingRight": 4,
                "paddingTop": 4,
              },
              {
                "flexDirection": "row",
              },
              {
                "alignSelf": "flex-start",
              },
              [
                {
                  "opacity": undefined,
                },
                [
                  {
                    "backgroundColor": "#1fff00",
                  },
                  {
                    "marginLeft": 8,
                    "marginTop": 8,
                  },
                  undefined,
                ],
              ],
            ]
          }
          testID="tag/Obj1"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-light",
                },
                {
                  "fontSize": 12,
                },
                {
                  "lineHeight": 16,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            Obj1
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "borderBottomLeftRadius": 9999,
                "borderBottomRightRadius": 9999,
                "borderTopLeftRadius": 9999,
                "borderTopRightRadius": 9999,
              },
              {
                "paddingBottom": 4,
                "paddingLeft": 4,
                "paddingRight": 4,
                "paddingTop": 4,
              },
              {
                "flexDirection": "row",
              },
              {
                "alignSelf": "flex-start",
              },
              [
                {
                  "opacity": undefined,
                },
                [
                  {
                    "backgroundColor": "#ff3f00",
                  },
                  {
                    "marginLeft": 8,
                    "marginTop": 8,
                  },
                  undefined,
                ],
              ],
            ]
          }
          testID="tag/Obj2"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-light",
                },
                {
                  "fontSize": 12,
                },
                {
                  "lineHeight": 16,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            Obj2
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "borderBottomLeftRadius": 9999,
                "borderBottomRightRadius": 9999,
                "borderTopLeftRadius": 9999,
                "borderTopRightRadius": 9999,
              },
              {
                "paddingBottom": 4,
                "paddingLeft": 4,
                "paddingRight": 4,
                "paddingTop": 4,
              },
              {
                "flexDirection": "row",
              },
              {
                "alignSelf": "flex-start",
              },
              [
                {
                  "opacity": undefined,
                },
                [
                  {
                    "backgroundColor": "#ffff00",
                  },
                  {
                    "marginLeft": 8,
                    "marginTop": 8,
                  },
                  undefined,
                ],
              ],
            ]
          }
          testID="tag/Obj3"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-light",
                },
                {
                  "fontSize": 12,
                },
                {
                  "lineHeight": 16,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            Obj3
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "borderBottomLeftRadius": 9999,
                "borderBottomRightRadius": 9999,
                "borderTopLeftRadius": 9999,
                "borderTopRightRadius": 9999,
              },
              {
                "paddingBottom": 4,
                "paddingLeft": 4,
                "paddingRight": 4,
                "paddingTop": 4,
              },
              {
                "flexDirection": "row",
              },
              {
                "alignSelf": "flex-start",
              },
              [
                {
                  "opacity": undefined,
                },
                [
                  {
                    "backgroundColor": "#100f50",
                  },
                  {
                    "marginLeft": 8,
                    "marginTop": 8,
                  },
                  undefined,
                ],
              ],
            ]
          }
          testID="tag/Obj4"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-light",
                },
                {
                  "fontSize": 12,
                },
                {
                  "lineHeight": 16,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            Obj4
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          [
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={5}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID="objectives/create"
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Medium",
              },
              {
                "fontSize": 14,
              },
              {
                "color": "#505050",
              },
              {
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Créer un objectif
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ObjectiveBottomSheet should render error 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "paddingLeft": 24,
          "paddingRight": 24,
        },
        {
          "paddingBottom": 16,
          "paddingTop": 16,
        },
        [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Bold",
          },
          {
            "fontSize": 20,
          },
          {
            "lineHeight": 28,
          },
          {
            "textAlign": "center",
          },
        ]
      }
    >
      Mes objectifs
    </Text>
    <View
      style={
        [
          {
            "marginBottom": 24,
            "marginTop": 24,
          },
          [
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View
        style={
          [
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "alignSelf": "center",
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      >
        <ViewManagerAdapter_ExpoImage
          contentFit="contain"
          contentPosition={
            {
              "left": "50%",
              "top": "50%",
            }
          }
          height={100}
          onError={[Function]}
          onLoad={[Function]}
          onLoadStart={[Function]}
          onProgress={[Function]}
          placeholder={[]}
          source={[]}
          style={
            {
              "height": 100,
              "width": 100,
            }
          }
          transition={null}
          width={100}
        />
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
              {
                "marginTop": 24,
              },
            ]
          }
        >
          Une erreur s'est produite
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          [
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={5}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID="objectives/create"
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Medium",
              },
              {
                "fontSize": 14,
              },
              {
                "color": "#505050",
              },
              {
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Créer un objectif
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ObjectiveBottomSheet should render loading 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "paddingLeft": 24,
          "paddingRight": 24,
        },
        {
          "paddingBottom": 16,
          "paddingTop": 16,
        },
        [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Bold",
          },
          {
            "fontSize": 20,
          },
          {
            "lineHeight": 28,
          },
          {
            "textAlign": "center",
          },
        ]
      }
    >
      Mes objectifs
    </Text>
    <View
      style={
        [
          {
            "marginBottom": 24,
            "marginTop": 24,
          },
          [
            undefined,
            undefined,
          ],
        ]
      }
    >
      <ActivityIndicator />
    </View>
    <View
      style={
        [
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          [
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={5}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID="objectives/create"
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Medium",
              },
              {
                "fontSize": 14,
              },
              {
                "color": "#505050",
              },
              {
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Créer un objectif
        </Text>
      </View>
    </View>
  </View>
</View>
`;
